File: src/SIStorage.Service.Contract/Models/WellKnownSIStorageServiceErrorCode.cs
namespace SIStorage.Service.Contract.Models;

/// <summary>
/// Defines well-known SIStorage service error codes.
/// </summary>
public enum WellKnownSIStorageServiceErrorCode
{
    /// <summary>
    /// File is empty.
    /// </summary>
    FileEmpty,

    /// <summary>
    /// File is too large.
    /// </summary>
    FileTooLarge,

    /// <summary>
    /// Multipart ContentType is required.
    /// </summary>
    MultipartContentTypeRequired,

    /// <summary>
    /// Content-Length header is required.
    /// </summary>
    ContentLengthHeaderRequired,

    /// <summary>
    /// Content-Disposition header is required.
    /// </summary>
    ContentDispositionHeaderRequired,

    /// <summary>
    /// Disposition FileName is required.
    /// </summary>
    DispositionFileNameRequired,

    /// <summary>
    /// Package not found.
    /// </summary>
    PackageNotFound,
}

File: web/src/models/WellKnownSIStorageServiceErrorCode.ts
/** Defines well-known SIStorage service error codes. */
const enum WellKnownSIStorageServiceErrorCode {
	/** File is empty. */
	FileEmpty,

	/** File is too large. */
	FileTooLarge,

	/** Multipart ContentType is required. */
	MultipartContentTypeRequired,

	/** Content-Length header is required. */
	ContentLengthHeaderRequired,

	/** Content-Disposition header is required. */
	ContentDispositionHeaderRequired,

	/** Disposition FileName is required. */
	DispositionFileNameRequired,

	/** Package not found. */
	PackageNotFound,
}

export default WellKnownSIStorageServiceErrorCode;